### PR TITLE
Add cloud init support

### PR DESCRIPTION
**Pull request by brandondunne**
Move cloud_init configuration from ManageIQ to ovirt gem.  Allow for both newer and older styles of attaching cloud_init data.

### CODE REVIEW

**Comment by brandondunne** [comment]
@Fryguy @gmcculloug Please review.

**Comment by brandondunne** [comment]
@Fryguy Any ideas on what to do with the test failures?  It's a difference in YAML.dump in the different versions of ruby.  I added a hack commit to get the tests passing, what are your thoughts?

File: lib/ovirt/vm.rb [hunk 1]
```diff
@@ -332,5 +352,19 @@ def create_template(options={})
...
+
+    private
+
+    def converted_cloud_init(content)
+      ovirt_cloud_init_keys = %w(active_directory_ou authorized_ssh_keys dns_search dns_servers domain host_name input_locale nic_configurations org_name regenerate_ssh_keys root_password system_locale timezone ui_language user_locale user_name)
+
+      require 'yaml'
+      raw_content = YAML.load(content)
+
+      hash = ovirt_cloud_init_keys.each_with_object({}) { |k, h| h[k] = raw_content.delete(k) }
```
Comment (Fryguy) [inline lib/ovirt/vm.rb]: Can this just be simplified to `content.slice(ovirt_cloud_init_keys)`?

File: spec/factories/vm.rb [hunk 2]
```diff
@@ -1,7 +1,12 @@
 FactoryGirl.define do
-  factory :vm_full, :class => "Ovirt::Vm" do
+  factory :vm, :class => "Ovirt::Vm" do
+    service { build(:service) }
+    initialize_with { new(service, {}) }
```
Comment (Fryguy) [inline spec/factories/vm.rb]: Can you flip these two lines?  It's cleare to me when the intializer comes first before specific attributes.

**Comment by Fryguy** [comment]
> Any ideas on what to do with the test failures? It's a difference in YAML.dump in the different versions of ruby. I added a hack commit to get the tests passing, what are your thoughts?

Is it because you are essentially double embedding?  Would it work if you moved it to a separate yml file that you read in?

File: lib/ovirt/vm.rb (hunk 1 above)
Comment (brandondunne) [inline lib/ovirt/vm.rb]: We want the ovirt specific keys to be removed from the raw_content hash, so we would have to do something like:
```ruby
raw_content = raw_content.slice!(ovirt_cloud_init_keys)
```
I think Hash#delete is less confusing in this case.

**Comment by brandondunne** [comment]
@Fryguy It's a hash with long string values that gets YAML dumped.  The string values appear to be escaped or not based on the ruby version.  The dumped YAML string is then set as a value in the XML.  (Ovirt +cloudinit works with or without the extra quotes)

**Comment by brandondunne** [comment]
@Fryguy I changed the test data to not have anything that will get quoted, problem solved

File: lib/ovirt/vm.rb (hunk 1 above)
Comment (Fryguy) [inline lib/ovirt/vm.rb]: I see...Yeah, all the approaches are equal then.

Comment (Fryguy) [inline lib/ovirt/vm.rb]: Actually..wait...why do you have to remove the keys from raw_content?  You don't even return it.

**Comment by Fryguy** [comment]
:+1: will merge when green.

**Pull request merged**